### PR TITLE
LPD-34301 Keep the value as it came

### PR DESCRIPTION
**Pull request by p-albuquerque**
Related issue: https://issues.liferay.com/browse/LPD-34301

### CODE REVIEW

**Comment by carolmariaabb** [comment]
@p-albuquerque could you please add a simple test on our side? I know it's already covered by another test, but it would be good to have a related test running in our daily suite.

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java [hunk 1]
```diff
@@ -4030,7 +4030,8 @@ else if (sqlType == Types.INTEGER) {
 			preparedStatement.setInt(index, GetterUtil.getInteger(value));
 		}
 		else if (sqlType == Types.VARCHAR) {
-			preparedStatement.setString(index, String.valueOf(value));
+			preparedStatement.setString(
+				index, GetterUtil.getString(value, null));
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:4034]: @p-albuquerque I've never seen `GetterUtil` being used with `null` as the default value.. 

Did you see some other example in our code? What if you do `(String)value` instead?

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java [hunk 2]
```diff
@@ -3066,6 +3066,8 @@ public void testUpdateObjectEntry() throws Exception {
 				"lastName", "o Discípulo Amado"
 			).put(
 				"listTypeEntryKeyRequired", "listTypeEntryKey2"
+			).put(
+				"middleName", () -> null
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java:3070]: @p-albuquerque wasn't the value of this field already `null`? If so, how this test was passing?

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Comment by carolmariaabb** [comment]
ci:test:object

**Comment by carolmariaabb** [comment]
ci:help

**Comment by carolmariaabb** [comment]
ci:test:object-db-integration

**Review by carolmariaabb — APPROVED** [review]

**Comment by p-albuquerque** [comment]
ci:test:headless

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java [hunk 3]
```diff
@@ -4162,7 +4162,7 @@ else if (sqlType == Types.INTEGER) {
 				insertedValues, columnNames.get(index - 1), integer);
 		}
 		else if (sqlType == Types.VARCHAR) {
-			String string = String.valueOf(value);
+			String string = (String)value;
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:4165]: @p-albuquerque can you test if this solution works when we have a text field and send a numeric value to the field?

Comment (p-albuquerque) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:4165]: works as expected

**Comment by p-albuquerque** [comment]
The failures [here](https://testray.liferay.com/#/project/34754/routines/34756/build/71688483/case-result/71696267) is caused by the changes. I'm analyzing how to overcome the situation

**Comment by p-albuquerque** [comment]
The failures [here](https://testray.liferay.com/#/project/34754/routines/34756/build/71688483/case-result/71696267) is related to my changes. I'm analyzing how to overcome them

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:sf

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java [hunk 4]
```diff
@@ -4162,7 +4162,14 @@ else if (sqlType == Types.INTEGER) {
...
 		else if (sqlType == Types.VARCHAR) {
-			String string = String.valueOf(value);
+			String string;
+
+			if (value == null) {
+				string = null;
+			}
+			else {
+				string = String.valueOf(value);
+			}
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:4172]: @p-albuquerque can we simplify?

```
String string = null;

if (value != null) {
   string = String.valueOf(value);
}
```

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:headless

**Comment by p-albuquerque** [comment]
ci:test:object

**Comment by p-albuquerque** [comment]
ci:test:headless

**Comment by magjed4289** [comment]
@p-albuquerque from the Headless end the API Builder tests are passing correctly (the failures are unrelated to your PR). Thank you!

**Comment by p-albuquerque** [comment]
@magjed4289 thank you!

**Comment by p-albuquerque** [comment]
ci:forward:force

**Comment by marcelabc** [comment]
ci:forward:force

**Pull request closed without merging**